python/rascaline/rascaline/utils/clebsch_gordan/_cg_cache.py

```diff
@@ -181,85 +236,188 @@ def use_mops(self):
     def coeffs(self):
         return self._coeffs
 
-    @staticmethod
-    def build_coeff_dict(lambda_max: int, sparse: bool, use_mops: bool):
-        """
-        Builds a dictionary of Clebsch-Gordan coefficients for all possible
-        combination of l1 and l2, up to lambda_max.
-        """
-        # real-to-complex and complex-to-real transformations as matrices
-        r2c = {}
-        c2r = {}
-        coeff_dict = {}
-        for lambda_ in range(0, lambda_max + 1):
-            c2r[lambda_] = _complex2real(lambda_)
-            r2c[lambda_] = _real2complex(lambda_)
-
-        for l1 in range(lambda_max + 1):
-            for l2 in range(lambda_max + 1):
-                for lambda_ in range(
-                    max(l1, l2) - min(l1, l2), min(lambda_max, (l1 + l2)) + 1
-                ):
-                    complex_cg = _complex_clebsch_gordan_matrix(l1, l2, lambda_)
-
-                    real_cg = (r2c[l1].T @ complex_cg.reshape(2 * l1 + 1, -1)).reshape(
-                        complex_cg.shape
-                    )
 
-                    real_cg = real_cg.swapaxes(0, 1)
-                    real_cg = (r2c[l2].T @ real_cg.reshape(2 * l2 + 1, -1)).reshape(
-                        real_cg.shape
+def _build_cg_coeff_dict(
+    lambda_max: int, sparse: bool, use_mops: bool, use_torch: bool
+):
+    """
+    Builds a dictionary of Clebsch-Gordan coefficients for all possible
+    combination of l1 and l2, up to lambda_max.
+    """
+    # real-to-complex and complex-to-real transformations as matrices
+    r2c: Dict[int, Array] = {}
+    c2r: Dict[int, Array] = {}
+
+    coeff_dict = {}
+
+    if use_torch:
+        complex_like = torch.empty(0, dtype=torch.complex128)
```

Comment:
Should these be dispatched or is it ok like this?

Reply:
This part should not be dispatched, since we have to determine somewhere what to take as reference to dispatch. The logic is:
> `DensityCorrelations` ---passes `arrays_backend` as `use_torch` argument--> `ClebschGordanReal`.


You could also change the API that `ClebschGordanReal.__init__` takes `like_array: Array` argument and then this can be dispatched, but having a `like_array` as input argument is not so good API I think.

Reply:
Ok noted thanks!